gatsby-node.js

```diff
@@ -62,75 +64,146 @@ exports.createPages = async (createPagesArgs) => {
+    );
+    const jobsXmlResponse = await fetch(PERSONIO_JOBS_URL_LANG);
+    const jobsXml = await jobsXmlResponse.text();
+    const options = {
+      tagValueProcessor: (a) => decode(a), // &#039; -> '
+    };
+    const jobsParse = xmlParser.parse(jobsXml, options);
+    const positions = jobsParse['workzag-jobs'].position;
+    return { positions, langKey };
+  });
 
-  positions.forEach((position) => {
-    position.satellytesPath = `/career/${position.id}-${slugify(position.name, {
-      lower: true,
-    })}/`;
-
-    // we need to normalize the jobDescription to an array, otherwise gatsby
-    // complains about different types for the same variable
-    position.jobDescriptions.jobDescription = Array.isArray(
-      position.jobDescriptions.jobDescription,
-    )
-      ? position.jobDescriptions.jobDescription
-      : [position.jobDescriptions.jobDescription];
-
-    position.jobDescriptions.jobDescription =
-      position.jobDescriptions.jobDescription.map((description) => {
-        return {
-          name: description.name.trim(),
-          value: description.value.trim(),
-        };
-      });
+  await Promise.all(positionsGroupedByLanguage).then(async (values) => {
+    for (let i = 0; i < values.length; i++) {
+      const positions = values[i].positions;
+      const langKey = values[i].langKey;
+
+      // position is valid if it has a job description and slug
+      const isValidPosition = (position) => {
+        return (
+          position.jobDescriptions.jobDescription &&
+          Array.isArray(position.jobDescriptions.jobDescription)
+        );
+      };
+
+      const getSlug = (position) => {
+        if (position.satellytesPath) return position.satellytesPath;
+        const slug = position.jobDescriptions.jobDescription.find(
+          (description) =>
+            description.name.trim() === 'Slug' && description.value.trim(),
+        );
+        return `/career/${slug.value.trim()}/`;
+      };
+
+      positions.forEach((position) => {
+        if (!isValidPosition(position)) {
+          return null;
+        }
 
-    const description = position.jobDescriptions.jobDescription.find(
-      (description) => description.name === PERSONIO_SHORT_DESCRIPTION_NAME,
-    );
-    if (!description || !description.value) {
-      console.warn(
-        `No description for job "${position.name}" (${position.id}) found!`,
-      );
-    }
+        // we need to normalize the jobDescription to an array, otherwise gatsby
+        // complains about different types for the same variable
+        position.jobDescriptions.jobDescription = Array.isArray(
+          position.jobDescriptions.jobDescription,
+        )
+          ? position.jobDescriptions.jobDescription
+          : [position.jobDescriptions.jobDescription];
+
+        position.jobDescriptions.jobDescription =
+          position.jobDescriptions.jobDescription
+            .map((description) => {
+              if (description.name.trim() === 'Slug') {
```

Comment:
Like `PERSONIO_SHORT_DESCRIPTION_NAME`, the `Slug` should be put into a variable.